gridcomps/History/MAPL_StationSamplerMod.F90

```diff
@@ -0,0 +1,461 @@
+                typekind=ESMF_TYPEKIND_R4,ungriddedLBound=lb,ungriddedUBound=ub,_RC)
+           call ESMF_FieldGet(dst_field,farrayptr=p_dst_3d,_RC)
+           !!if (this%vdata%regrid_type==VERTICAL_METHOD_ETA2LEV) then
+           !!   allocate(p_new_lev(size(p_src_3d,1),size(p_src_3d,2),this%vdata%lm),_STAT)
+           !!   call this%vdata%regrid_eta_to_pressure(p_src_3d,p_new_lev,_RC)
+           !!   call this%regridder%regrid(p_new_lev,p_dst_3d,_RC)
+           !!else
+           call this%regridder%regrid(p_src_3d,p_dst_3d,_RC)
+           !!end if
+           !! 
+           if (mapl_am_i_root()) then
+              nx=size(p_dst_3d,1); nz=size(p_dst_3d,2); allocate(arr(nz, nx))
+              arr=reshape(p_dst_3d,[nz,nx],order=[2,1])
+              call this%formatter%put_var(xname,arr,&
+                   start=[1,1,this%obs_written],count=[nz,nx,1],_RC)
+              !note:     lev,station,time
+           end if
+           call ESMF_FieldDestroy(dst_field,nogarbage=.true.)
+        else
+           _FAIL('grid2LS regridder: rank > 3 not implemented')
+        end if  ! rank
+     enddo      ! fieldCount
+     _RETURN(_SUCCESS)
+   end subroutine append_file
+
+   
+   subroutine create_file_handle(this,filename,rc)
+     class(StationSampler), intent(inout) :: this
+     character(len=*), intent(inout) :: filename  ! for ouput nc
+     integer, optional, intent(out) :: rc
+     type(variable) :: v
+     integer :: status
+
+     this%ofile = trim(filename)
+     v = this%time_info%define_time_variable(_RC)
+     call this%fmd%modify_variable('time',v,_RC)
+     !
+     if (mapl_am_I_root()) then
+        call this%formatter%create(trim(filename),_RC)
+        call this%formatter%write(this%fmd,_RC)
+        call this%formatter%put_var('longitude',this%lons,_RC)
+        call this%formatter%put_var('latitude',this%lats,_RC)
+        call this%formatter%put_var('station_id',this%station_id,_RC)                
+     end if
+     this%obs_written = 0
+     _RETURN(_SUCCESS)
+   end subroutine create_file_handle
+
+  
+   subroutine close_file_handle(this,rc)
+     class(StationSampler), intent(inout) :: this
+     integer, optional, intent(out) :: rc
+     integer :: status
+     if (trim(this%ofile) /= '') then
+        if (mapl_am_i_root()) then
+           call this%formatter%close(_RC)
+        end if
+     end if
+     _RETURN(_SUCCESS)
+   end subroutine close_file_handle
+   
+
+   function compute_time_for_current(this,current_time,rc) result(rtimes)
+     class(StationSampler), intent(inout) :: this
+     type(ESMF_Time), intent(in) :: current_time
+     integer, optional, intent(out) :: rc
+     real(ESMF_KIND_R8), allocatable :: rtimes(:)
+     integer :: i,status
+     type(ESMF_TimeInterval) :: tint
+     type(ESMF_Time) :: file_start_time
+     character(len=ESMF_MAXSTR) :: tunits
+     
+     allocate(rtimes(1),_STAT)
+     call this%get_file_start_time(file_start_time,tunits,_RC)
```

Comment:
```suggestion
     call this%get_file_start_time(file_start_time,tunit,_RC)
```
I think it should be `tunit` rather than `tunits`, but maybe I've not looked closely enough.  Scalars should be singular nouns.

Reply:
I agree. I also felt a bit unease when adopting it from MAPL_HistoryTrajectoryMod.F90, where the tunits was an output from subroutine which process the Time-units [day,month,year] from a string in NC to yield a unique time-unit.